bgpd/bgp_ecommunity.c

```diff
@@ -601,9 +601,12 @@ ecommunity_str2com (const char *str, int type, int keyword_included)
    ECOMMUNITY_FORMAT_ROUTE_MAP
    ECOMMUNITY_FORMAT_COMMUNITY_LIST
    ECOMMUNITY_FORMAT_DISPLAY
+
+   Filter is added to display only ECOMMUNITY_ROUTE_TARGET in some cases. 
+   0 value displays all
 */
 char *
-ecommunity_ecom2str (struct ecommunity *ecom, int format)
+ecommunity_ecom2str (struct ecommunity *ecom, int format, int filter)
```

Comment:
in regression valgrind reporting that this routine is now returning an uninitialized value.
==341== 16 errors in context 1 of 1:
==341== Conditional jump or move depends on uninitialised value(s)
==341==    at 0x34E0247E2C: vfprintf (in /lib64/libc-2.12.so)
==341==    by 0x34E026F751: vsnprintf (in /lib64/libc-2.12.so)
==341==    by 0x52EB9C: vty_out (vty.c:115)
==341==    by 0x4F56EB: rfapi_vty_out_vncinfo (rfapi_vty.c:475)
==341==    by 0x45C565: route_vty_out (bgp_route.c:6305)
==341==    by 0x487A17: bgp_show_mpls_vpn (bgp_mplsvpn.c:731)
==341==    by 0x462B55: show_ip_bgp (bgp_route.c:8425)
==341==    by 0x536DC3: cmd_execute_command_real (command.c:834)
==341==    by 0x536F04: cmd_execute_command (command.c:888)
==341==    by 0x52F86F: vty_command (vty.c:457)
==341==    by 0x531675: vty_execute (vty.c:1348)
==341==    by 0x531C54: vty_read (vty.c:1576)
==341== 
{
   <insert_a_suppression_name_here>
   Memcheck:Cond
   fun:vfprintf
   fun:vsnprintf
   fun:vty_out
   fun:rfapi_vty_out_vncinfo
   fun:route_vty_out
   fun:bgp_show_mpls_vpn
   fun:show_ip_bgp
   fun:cmd_execute_command_real
   fun:cmd_execute_command
   fun:vty_command
   fun:vty_execute
   fun:vty_read